### PR TITLE
Fix indexing bug

### PR DESCRIPTION
**Pull request by mnasty**
Fix playlist indexing bug detailed in issue #99 

Please review, test and merge

Reach out with concerns :)

### CODE REVIEW

**Comment by StefanLobbenmeier** [comment]
Doing some testing, this is the playlist I used:

Output Template I used: %(playlist)s/%(playlist_index)s - %(title)s.%(ext)s

Result when the videos are split (<50):
image

Result with splitting disabled ("Never split"):
image

So looking very nice 😄 thank you very much

**Comment by StefanLobbenmeier** [comment]
Regarding failing tests - weird that the CI is broken as well (maybe the node version was deprecated)

File: modules/Utils.js [hunk 1]
```diff
@@ -206,11 +206,7 @@ class Utils {
         if(metadata == null) return null;
         for(const video of metadata) {
             if(video.video_url === video_url) {
-                if(playlist_url == null) {
-                    return video;
-                } else if(playlist_url === video.playlist_url) {
-                    return video;
-                }
+                return video;
```
Comment (StefanLobbenmeier) [inline modules/Utils.js:209]: actually those changes break 2 new tests, can you tell me why they are required? then we can adjust the expectation or remove the tests

File: modules/Utils.js [hunk 2]
```diff
@@ -179,7 +179,7 @@ class Utils {
         for(const entry of query.entries) {
             let url;
             if (entry.url == null) url = entry.webpage_url;
-            else url = (entry.ie_key != null && entry.ie_key === "Youtube") ? "https://youtube.com/watch?v=" + entry.url : entry.url;
+            else url = entry.url;
```
Comment (StefanLobbenmeier) [inline modules/Utils.js:182]: (this also breaks a test)

File: modules/Utils.js (hunk 1 above)
Comment (mnasty) [reply modules/Utils.js:209]: Yes, without the additional return statement the playlist metadata is not made available at the time it is consumed by the individual threads that are downloading the video. I have no idea why there was a conditional statement there at all as it appears to have no effect. Of course there is no documentation or comments to refer to..

You can try removing the return statement and you will see the old behavior repeat itself. None of the tests failed for me,  can you refer to which tests are failing so I can investigate further?

Thanks!

File: modules/Utils.js (hunk 2 above)
Comment (mnasty) [reply modules/Utils.js:182]: This appears to be a mistake by the developer. If you log the input to the ternary statement you will see that it doesn't work as intended. The url is duplicated for the reproducible test case that I provided.. which means it is effectively always false.

Again, this statement appeared to have no effect so if you can link the failing test I will double check on it.

**Comment by StefanLobbenmeier** [comment]
Ignoring the test failures for now - it’s verified through manual test anyway so LGTM

**Pull request merged**